examples/permissioned-token/test/common.js

```diff
@@ -0,0 +1,379 @@
+function syncAccounts (runtime) {
+  master = runtime.getAccount(master.address);
+  alice = runtime.getAccount(alice.address);
+  bob = runtime.getAccount(bob.address);
+  elon = runtime.getAccount(elon.address);
+}
+
+/**
+ * - Setup token (ASA gold)
+ * - Setup Controller asc
+ * - Setup asset clawback + update asset clawback to escrow contract
+ * - Setup permissions smart contract
+ * NOTE: During setup - ASA.reserve, ASA.manager & current_permissions_manager is set as alice.address
+ */
+function setupEnv () {
+  // Create Accounts and Env
+  alice = new AccountStore(minBalance, { addr: ALICE_ADDRESS, sk: new Uint8Array(0) });
+  bob = new AccountStore(minBalance);
+  elon = new AccountStore(minBalance);
+  runtime = new Runtime([master, alice, bob, elon]);
+
+  // Deploy ASA
+  assetIndex = runtime.addAsset('gold', { creator: { ...alice.account, name: 'alice' } });
+  const asaDef = runtime.getAssetDef(assetIndex);
+
+  // Deploy Controller SSC
+  let sscFlags = {
+    sender: alice.account,
+    localInts: 0,
+    localBytes: 0,
+    globalInts: 2,
+    globalBytes: 0,
+    foreignAssets: [assetIndex]
+  };
+  CONTROLLER_PROGRAM = getProgram(CONTROLLER, { TOKEN_ID: assetIndex });
+  controllerAppID = runtime.addApp(
+    sscFlags, {}, CONTROLLER_PROGRAM, CLEAR_STATE_PROGRAM
+  );
+
+  // Deploy Clawback Lsig and Modify Asset
+  CLAWBACK_PROGRAM = getProgram(CLAWBACK, {
+    TOKEN_ID: assetIndex,
+    CONTROLLER_APP_ID: controllerAppID
+  });
+  lsig = runtime.getLogicSig(CLAWBACK_PROGRAM, []);
+  fund(runtime, master, lsig.address());
+  runtime.executeTx({
+    type: types.TransactionType.ModifyAsset,
+    sign: types.SignType.SecretKey,
+    fromAccount: alice.account,
+    assetID: assetIndex,
+    fields: {
+      manager: asaDef.manager,
+      reserve: asaDef.reserve,
+      freeze: asaDef.freeze,
+      clawback: lsig.address()
+    },
+    payFlags: { totalFee: 1000 }
+  });
+  optInToASA(runtime, lsig.address(), assetIndex);
+
+  syncAccounts(runtime);
+
+  // Deploy Permissions SSC
+  PERMISSIONS_PROGRAM = getProgram(PERMISSIONS, { PERM_MANAGER: alice.address });
+  sscFlags = {
+    sender: alice.account,
+    localInts: 1,
+    localBytes: 0,
+    globalInts: 2,
+    globalBytes: 1,
+    appArgs: [`int:${controllerAppID}`]
+  };
+  permissionsAppId = runtime.addApp(
+    sscFlags, {}, PERMISSIONS_PROGRAM, CLEAR_STATE_PROGRAM
+  );
+
+  // Add permissions SSC config to Controller SSC
+  const appArgs = [
+    'str:set_permission',
+    `int:${permissionsAppId}`
+  ];
+  runtime.executeTx({
+    type: types.TransactionType.CallNoOpSSC,
+    sign: types.SignType.SecretKey,
+    fromAccount: alice.account,
+    appId: controllerAppID,
+    payFlags: { totalFee: 1000 },
+    appArgs: appArgs,
+    foreignAssets: [assetIndex]
+  });
+
+  // Refresh Accounts
+  syncAccounts(runtime);
+
+  return [
+    runtime,
+    master,
+    alice,
+    bob,
+    elon,
+    assetIndex,
+    controllerAppID,
+    lsig,
+    permissionsAppId
+  ];
```

Comment:
I think we should have a class  for this:
```
class Context:
  let runtime: Runtime
  let alice: ...

  constructor(runtime, alice, ...) ...
  ... 

  syncAccounts( { .... } )
}
```
This function can create a new Context and return it.
This way we don't have a flying global variables, and we encapsulate the runtime and the required accounts.

Reply:
ok

Reply:
done